packages/yarnpkg-fslib/sources/path.ts

```diff
@@ -58,6 +70,27 @@ npath.toPortablePath = toPortablePath;
 npath.contains = (from: NativePath, to: NativePath) => contains(npath, from, to);
 ppath.contains = (from: PortablePath, to: PortablePath) => contains(ppath, from, to);
 
+npath.FileURL = NativeFileURL;
+ppath.FileURL = PortableFileURL;
+
+npath.fromFileURL = (url: FileURL<NativePath>) => fileURLToPath(url);
+ppath.fromFileURL = (url: FileURL<PortablePath>) => fileURLToPosixPath(url);
+
+npath.toFileURL = (p: NativePath) => pathToFileURL(p);
+ppath.toFileURL = (p: PortablePath) => posixPathToFileURL(p);
+
+npath.PathBuffer = NativePathBuffer;
+ppath.PathBuffer = PortablePathBuffer;
+
+npath.fromPathBuffer = (buffer: PathBuffer<NativePath>) => buffer.toString();
+ppath.fromPathBuffer = (buffer: PathBuffer<PortablePath>) => buffer.toString();
```

Comment:
Shouldn't that be `buffer.toString('binary')`? Per doc:

> Paths specified using a Buffer are useful primarily on certain POSIX operating systems that treat file paths as opaque byte sequences. On such systems, it is possible for a single file path to contain sub-sequences that use multiple character encodings. As with string paths, Buffer paths may be relative or absolute:

Without explicit encoding, I think the buffer is interpreted as UTF8.